com.sap.cloud.lm.sl.cf.core/src/main/java/com/sap/cloud/lm/sl/cf/core/cf/v1/ResourceType.java

```diff
@@ -1,32 +1,50 @@
 package com.sap.cloud.lm.sl.cf.core.cf.v1;
 
+import java.util.Collections;
 import java.util.EnumSet;
+import java.util.HashSet;
 import java.util.Set;
 
+import com.sap.cloud.lm.sl.cf.core.model.SupportedParameters;
+
 public enum ResourceType {
-    MANAGED_SERVICE("managed-service"), USER_PROVIDED_SERVICE("user-provided-service"), EXISTING_SERVICE(
-        "existing-service"), EXISTING_SERVICE_KEY("existing-service-key");
+    MANAGED_SERVICE("managed-service", SupportedParameters.SERVICE, SupportedParameters.SERVICE_PLAN), USER_PROVIDED_SERVICE(
+        "user-provided-service",
+        SupportedParameters.SERVICE_CONFIG), EXISTING_SERVICE("existing-service"), EXISTING_SERVICE_KEY("existing-service-key");
 
-    private String value;
+    private String resourceName;

```

Comment:
Call it just "name". The name of the type is different than the name of the resource.